app/code/core/Mage/Sales/Model/Observer.php

```diff
@@ -60,7 +60,7 @@ public function cleanExpiredQuotes($schedule)
 
             $quotes->addFieldToFilter('store_id', $storeId);
             $quotes->addFieldToFilter('updated_at', array('to'=>date("Y-m-d", time()-$lifetime)));
-            $quotes->addFieldToFilter('is_active', 0);
+            $quotes->addFieldToFilter('is_active', 1);
 
             foreach ($this->getExpireQuotesAdditionalFilterFields() as $field => $condition) {
```

Comment:
Note2: We can also remove this "useless" foreach

Reply:
Removing may break BC.